File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def dfs(self, root):
        if root:
            self.dfs(root.left)
            self.container.append(root.val)
            self.dfs(root.right)
    
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.container = []  
        self.dfs(root)  
        return self.container[k-1]  
        

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private ArrayList<Integer> list=new ArrayList();
    private void dfs(TreeNode root){
        if(root!=null){
            dfs(root.left);
            list.add(root.val);
            dfs(root.right);
        }
    }
    public int kthSmallest(TreeNode root, int k) {
        dfs(root);
        return list.get(k-1);
    }
}